apps/builder/app/shared/remix/handle-request.server.tsx

```diff
@@ -13,10 +22,33 @@ export const handleRequest = (
     <RemixServer context={remixContext} url={request.url} />
   );
 
+  const head = renderHeadToString({ request, remixContext, Head });
+
   responseHeaders.set("Content-Type", "text/html");
   responseHeaders.set("Accept-CH", "Sec-CH-Prefers-Color-Scheme");
 
-  return new Response("<!DOCTYPE html>" + insertCriticalCss(markup), {
+  // data-builder="true" to distinguish between builder and canvas at the entry.client.tsx
+  // inject fonts and base body styles here instead of `export links = ` to avoid FOUC
+  const body = `
+    <!DOCTYPE html>
+    <html lang="en" data-builder="true">
```

Comment:
I feel like this is all leaky abstraction, we shouldn't have app specific stuff in the layout we are reusing across different apps

Reply:
effectively you are writing the app name inside the layout

Reply:
could be data-app="builder" and consistent, so all apps would be there, but even then, its still not great, I bet we can avoid it all